src/Microsoft.OData.Client/ALinq/ResourceBinder.GroupByResultSelectorRewriter.cs

```diff
@@ -0,0 +1,663 @@
+                    case SequenceMethod.SumNullableDoubleSelector:
+                    case SequenceMethod.SumNullableDecimalSelector:
+                    case SequenceMethod.SumNullableLongSelector:
+                    case SequenceMethod.SumNullableSingleSelector:
+                    case SequenceMethod.AverageIntSelector:
+                    case SequenceMethod.AverageDoubleSelector:
+                    case SequenceMethod.AverageDecimalSelector:
+                    case SequenceMethod.AverageLongSelector:
+                    case SequenceMethod.AverageSingleSelector:
+                    case SequenceMethod.AverageNullableIntSelector:
+                    case SequenceMethod.AverageNullableDoubleSelector:
+                    case SequenceMethod.AverageNullableDecimalSelector:
+                    case SequenceMethod.AverageNullableLongSelector:
+                    case SequenceMethod.AverageNullableSingleSelector:
+                    case SequenceMethod.MinIntSelector:
+                    case SequenceMethod.MinDoubleSelector:
+                    case SequenceMethod.MinDecimalSelector:
+                    case SequenceMethod.MinLongSelector:
+                    case SequenceMethod.MinSingleSelector:
+                    case SequenceMethod.MinNullableIntSelector:
+                    case SequenceMethod.MinNullableDoubleSelector:
+                    case SequenceMethod.MinNullableDecimalSelector:
+                    case SequenceMethod.MinNullableLongSelector:
+                    case SequenceMethod.MinNullableSingleSelector:
+                    case SequenceMethod.MaxIntSelector:
+                    case SequenceMethod.MaxDoubleSelector:
+                    case SequenceMethod.MaxDecimalSelector:
+                    case SequenceMethod.MaxLongSelector:
+                    case SequenceMethod.MaxSingleSelector:
+                    case SequenceMethod.MaxNullableIntSelector:
+                    case SequenceMethod.MaxNullableDoubleSelector:
+                    case SequenceMethod.MaxNullableDecimalSelector:
+                    case SequenceMethod.MaxNullableLongSelector:
+                    case SequenceMethod.MaxNullableSingleSelector:
+                    case SequenceMethod.Count:
+                    case SequenceMethod.LongCount:
+                    case SequenceMethod.CountDistinctSelector:
+                        MemberInfo memberInfo;
+                        if (this.expressionsMap.TryGetValue(m, out memberInfo))
+                        {
+                            // Project plan compiler wouldn't know how to handle an aggregation method
+                            // Replace method call with a property access expression
+                            return Expression.PropertyOrField(this.lambdaParameter, memberInfo.Name);
+                        }
+                        // Something would have to be wrong for us to find ourselves here
+                        throw Error.NotSupported();

```

Comment:
Could we have some meaningful output string here for debug purposes or is this the case where it is hard to determine what a useful message would be.

Reply:
Done